engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -278,4 +284,63 @@ public static synchronized void changeLoggersLevel(@Nonnull Map<String, String>
     }
     return ImmutableMap.copyOf(matchedClasses);
   }
+
+  /**
+   * Returns logger log file name
+   * @return logFileName
+   */
+  @Evolving
+  public static String getLogFileName()
+  {
+    Set<FileAppender> fileAppenders = getRootFileAppenders();
+    //skip fetching log file name if we have multiple file Appenders
+    if (fileAppenders.size() == 1) {
+      FileAppender fileAppender = fileAppenders.iterator().next();
+      if (isErrorLevelEnable(fileAppender)) {
+        return fileAppender.getFile();
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Returns offset of current log file.
+   * @return logFileOffset
+   */
+  @Evolving
+  public static long getLogFileOffset()
+  {
+    Set<FileAppender> fileAppenders = getRootFileAppenders();
+    //skip fetching log file offset if we have multiple file Appenders
+    if (fileAppenders.size() == 1) {
+      FileAppender fileAppender = fileAppenders.iterator().next();
+      if (isErrorLevelEnable(fileAppender)) {
+        File logFile = new File(fileAppender.getFile());
+        return logFile.length();
+      }
+    }
+    return 0;
+  }
+
+  private static Set<FileAppender> getRootFileAppenders()
```

Comment:
@vlad one problem I see with this approach, if we update appender properties after app lauch, it won't be reflected in our code as we are doing static one time loading of appender. Say after some time someone sets immediateFlush=false or chagnes fileName, we wont' be able to pick those changes.